kube-core/src/metadata.rs

```diff
@@ -23,39 +23,57 @@ pub struct TypeMeta {
 /// schema without knowing the details of the version.
 #[derive(Deserialize, Serialize, Clone, Default, Debug)]
 #[serde(rename_all = "camelCase")]
-pub struct PartialObjectMeta {
+pub struct PartialObjectMeta<K> {
```

Comment:
```suggestion
pub struct PartialObjectMeta<K = DynamicObject> {
```

Not sure if we want to default to `DynamicObject` (for compat) but it's an option.

Reply:
I think it's a nice ergonomic default even by itself.